cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -686,6 +691,8 @@ void deleteApp(Id.Application appId, ApplicationSpecification spec) throws Excep
     }
     deleteProgramLocations(appId);
 
+    deleteAppBusinessMetadata(appId);
```

Comment:
there's an upgrade test in ApplicationLifecycleServiceTest, but I'm guessing it already sets up the business mds.  I think there are some integration tests but those aren't in the cdap repo.